addons/im_livechat/models/chatbot_script.py

```diff
@@ -182,19 +183,20 @@ def action_view_livechat_channels(self):
     # Tooling / Misc
     # --------------------------
 
-    def _format_for_frontend(self):
-        """ Small utility method that formats the script into a dict usable by the frontend code. """
-        self.ensure_one()
-
-        return {
-            'id': self.id,
-            'name': self.title,
-            'partner': {'id': self.operator_partner_id.id, 'type': 'partner', 'name': self.operator_partner_id.name},
-            'welcomeSteps': [
-                step._format_for_frontend()
-                for step in self._get_welcome_steps()
-            ]
-        }
+    def _to_store(self, store: Store, /, *, fields=None, **kwargs):
+        if fields is None:
+            fields = ["name", "partner", "welcome_steps"]
+        for script in self:
+            data = script._read_format(
+                [f for f in fields if f not in {"name", "partner", "welcome_steps"}], load=False
+            )[0]
+            if "name" in fields:
+                data["name"] = script.title
+            if "partner" in fields:
+                data["partner"] = Store.one(script.operator_partner_id, fields=["name"])
+            if "welcome_steps" in fields:
+                data["welcomeSteps"] = Store.many(script._get_welcome_steps())
```

Comment:
We're trying to remove the renames in the code, it might be better to change the incorrect field name in JS and keep the python name instead.